### PR TITLE
integration/TestDaemonProxy: Remove OTEL span

### PR DESCRIPTION
**Pull request by vvoland**
- extracted from: https://github.com/moby/moby/pull/47364

Don't use OTEL tracing in this test because we're testing the HTTP proxy behavior here and we don't want OTEL to interfere.

Fixes random `TestDaemonProxy` failures:
```
=== CONT  TestDaemonProxy/command-line_options
    daemon_test.go:280: assertion failed: 172.17.0.1:4318 (received string) != example.org:5001 (string): should not have used proxy
--- FAIL: TestDaemonProxy (0.50s)
    --- PASS: TestDaemonProxy/conflicting_options (0.50s)
    --- PASS: TestDaemonProxy/reload_sanitized (1.02s)
    --- PASS: TestDaemonProxy/configuration_file (1.25s)
    --- PASS: TestDaemonProxy/environment_variables (1.31s)
    --- FAIL: TestDaemonProxy/command-line_options (6.15s)
```

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: integration/daemon/daemon_test.go [hunk 1]
```diff
@@ -182,7 +183,9 @@ func TestConfigDaemonSeccompProfiles(t *testing.T) {
 func TestDaemonProxy(t *testing.T) {
 	skip.If(t, runtime.GOOS == "windows", "cannot start multiple daemons on windows")
 	skip.If(t, os.Getenv("DOCKER_ROOTLESS") != "", "cannot connect to localhost proxy in rootless environment")
-	ctx := testutil.StartSpan(baseContext, t)
+
+	// Don't setup OTEL here to avoid it hitting the HTTP proxy.
+	ctx := context.Background()
```
Comment (thaJeztah) [inline integration/daemon/daemon_test.go:188]: Curious; do we configure OTEL to be using HTTP? Or are we not configuring it to disable the "discovery" (`localhost:<some port>`)?

ISTR our intent was to disable the automatic discovery on localhost, but if it's still doing that, we may have missed something (or regressed somewhere)

Comment (vvoland) [reply integration/daemon/daemon_test.go:188]: Yes we use the OTEL http endpoint in CI: https://github.com/moby/moby/blob/9d1541526c5beab51d2344b47624a160f199bbe6/.github/actions/setup-tracing/action.yml#L13

The intent here is to not start any spans specifically for this test, even if OTLP exporter is configured.

**Comment by vvoland** [comment]
Failure is a known flaky test:
```
--- FAIL: TestLiveRestore (0.00s)
    --- SKIP: TestLiveRestore/autoremove (0.00s)
    --- FAIL: TestLiveRestore/volume_references (20.81s)
        --- PASS: TestLiveRestore/volume_references/restartPolicy (13.89s)
            --- PASS: TestLiveRestore/volume_references/restartPolicy/always (4.50s)
            --- PASS: TestLiveRestore/volume_references/restartPolicy/unless-stopped (4.58s)
            --- PASS: TestLiveRestore/volume_references/restartPolicy/on-failure (2.52s)
            --- PASS: TestLiveRestore/volume_references/restartPolicy/no (2.29s)
        --- FAIL: TestLiveRestore/volume_references/local_volume_with_mount_options (2.33s)
            --- SKIP: TestLiveRestore/volume_references/local_volume_with_mount_options/volume_still_mounted (0.00s)
        --- PASS: TestLiveRestore/volume_references/container_with_bind-mounts (2.30s)
```

**Review by cpuguy83 — APPROVED** [review]

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**